src/cachehash/main.py

```diff
@@ -48,6 +65,8 @@ def query(self, file_name: str, parameters=None, query: Union[str, None] = None)
 
     def hash_file(self, fp: Path) -> str:
         h = xh()
+        if not self.is_regular_file(fp):
+            return "Can't hash Unix Sockets"
```

Comment:
Could the user of cachehash use the hexdigest to query the db directly? If not, this function could return a BOOL, True for success, False for error.  Or at least "File at *file_path* not a valid file" since it could be directory, bad path etc, not necessarily a unix socket.

Reply:
Hmm, One idea that isn't super elegant would be to return a valid hash OR a string with a "error:" prefix.  That would allow the client to not to a getByHash("Can't hash Unix Sockets")

AWS CLI has a similar pattern and I hate it. It returns a result Obj OR an Error Obj and you have to do a klugey check.